### ISSUE TITLE
Wrong order tax amounts displayed when using specific tax configuration

### ISSUE THREAD

**Issue by jreinke** [magento/magento2#10347]
### Preconditions
1. Magento CE 2.1 with sample data installed
2. PHP 7 (does not matter IMHO)
3. MySQL 5.7 (does not matter IMHO)

### Steps to reproduce
1. Go to Magento admin
2. Create the following tax rates (Stores > Tax Zones and Rates):

Tax Identifier | Country | Zip | State | Rate
-------------- | ------- | --- | ----- | ----
Tax1 | France | * | * | 10%
Tax2 | France | * | * | 5%
Tax3 | France | * | * | 20%
3. Create the following tax rules (Stores > Tax Rules):

Name | Tax Rate | Customer Tax Class | Product Tax Class | Priority | Sort Order
---- | -------- | ------------------ | ----------------- | -------- | ----------
Tax1 | Tax1 | Retail Customer | Taxable Goods | 0 | 0
Tax2 | Tax2 | Retail Customer | Taxable Goods | 1 | 0
Tax3 | Tax3 | Retail Customer | Taxable Goods | 1 | 0
4. Enable full tax summary display in "Stores > Configuration > Sales > Tax > Orders, Invoices, Credit Memos Display Settings > Display Full Tax Summary"
5. Go to frontend and click on category "Gear > Bags"
6. Add product "Strive Shoulder Pack" to cart (qty = 1)
7. Go to checkout
8. Continue as guest or as authenticated user
9. Provide shipping address with country France (according to tax country configured in step 2)
10. Place order
11. Go to order details page

### Expected result
Tax total amount should be equal to the sum of the tax details amounts (12.00 for this use case)

### Actual result
Tax details sum is equal to 11.20
<img width="975" alt="magento2 issue tax details amounts" src="https://user-images.githubusercontent.com/598419/28633266-8576cdb6-7233-11e7-8625-795a84cb5a33.png">

Thanks!

**Comment by choukalos** [comment]
Questions:
- What country is this for?  
- Can you share details on what are the 3 taxes you need to collect?
- What is your tax configuration?  
- Are product prices loaded including tax?  
- What's your shipping origin set to?

Thanks!
Chuck

**Comment by jreinke** [comment]
Hi,

This is not for a specific purpose but this configuration comes with our internal test cases. France is just an example, you can test with another country if you want.

All other configurations are the Magento default ones. Nothing else has been modified. Just make a fresh install, follow the steps to reproduce and you should be able to reproduce the bug.

I had to debug this by myself and I think the bug may come from this code: https://github.com/magento/magento2/blob/99e85cbc45223baa3551e4c534b650c0d2c6358b/app/code/Magento/Tax/Model/Plugin/OrderSave.php#L98

`$percentSum` is never used while it should be instead of `$rates['percent']`.

A fix suggestion could look like this:
```
$percentSum = 0;
array_walk($taxRates, function($rate) use ($percentSum) {
    $percentSum += $rate['percent'];
});
foreach ($taxRates as $rate) {
    $realAmount = $rates['amount'] * $rate['percent'] / $percentSum;
    $realBaseAmount = $rates['base_amount'] * $rate['percent'] / $percentSum;
    $ratesIdQuoteItemId[$rates['id']][] = [
        'id' => $taxesArray['item_id'],
        'percent' => $rate['percent'],
        'code' => $rate['code'],
        'associated_item_id' => $taxesArray['associated_item_id'],
        'item_type' => $taxesArray['type'],
        'amount' => $rates['amount'],
        'base_amount' => $rates['base_amount'],
        'real_amount' => $realAmount,
        'real_base_amount' => $realBaseAmount,
    ];
}
```

Thanks

**Comment by jreinke** [comment]
Hi team,

Any feedback on this please?

**Comment by magento-engcom-team** [comment]
@jreinke, thank you for your report.
We've created internal ticket(s) MAGETWO-75545 to track progress on the issue.

**Comment by PieterCappelle** [comment]
Please assign this to me. I'll be working on this one, as I have the same problem in one of our projects.

**Comment by miguelbalparda** [comment]
Thanks a lot @PieterCappelle!!! There is a proposed fix for this @jreinke, can you check it and report back?

**Comment by okorshenko** [comment]
The issue has been fixed and delivered to 2.2-develop and 2.3-develop

**Comment by irshad-iflair** [comment]
@jreinke:  I am struggling for **how can I display separate tax rates on the product on selected stores** please refer to this. https://magento.stackexchange.com/questions/303861/magento-2-3-x-how-can-customise-tax-rules-for-different-stores-for-common-count
https://github.com/magento/magento2/issues/26831
